### PR TITLE
Enable test which now works because of serialization changes

### PR DESCRIPTION
**Pull request by holly-cummins**
https://github.com/quarkusio/quarkus/pull/35124 added several tests which could not pass, because of our [test classloading issues](https://github.com/quarkusio/quarkus/pull/34681). It's taken me a while (cough) to get on top of the code changes to make them pass, but @dmlloyd has fixed one category of them (and also https://github.com/quarkiverse/quarkus-pact/issues/73) with #40601. 

Because the test has been disabled for a while, it rotted slightly and got broken by some dev ui changes. The`DevUIJsonRPCTest` was seeing a non-default `quarkus.http.non-application-root-path`, so things failed. I can’t see where the problem value is coming from; perhaps some sticky config changes from a previous test? Cross-talk between tests is bad, but I didn’t want to get sucked into a config rabbit hole, so I just set my application to have the same config as the intruding config and moved on.

I also confirmed that #27821 and #22611 are still problems (and patched up their tests to fail for the right reasons again, so they will hopefully pass when the epic test classloading rewrite ETCR (https://github.com/quarkusio/quarkus/pull/34681) is done.

### CODE REVIEW

**Review by dmlloyd — APPROVED** [review]

**Comment by holly-cummins** [comment]
The failing tests are in my new test, so it looks like I need to be a bit more successful disabling it on native.

File: integration-tests/test-extension/extension-that-defines-junit-test-extensions/deployment/src/main/resources/application.properties [hunk 1]
```diff
@@ -1,193 +0,0 @@
-# Log settings
-# log level in lower case for testing
-quarkus.log.level=info
```
Comment (gsmet) [inline integration-tests/test-extension/extension-that-defines-junit-test-extensions/deployment/src/main/resources/application.properties:3]: Why is this file removed? It was useless?

Comment (holly-cummins) [reply integration-tests/test-extension/extension-that-defines-junit-test-extensions/deployment/src/main/resources/application.properties:3]: When I initially wrote these tests, I'd copied the files from another test. It had a whole bunch of settings that we didn't actually want in this test, such as setting the non-application-root to 1234. (Although, as mentioned above, even without the settings file, the non-application root is *still* 1234, which I haven't managed to explain.)

I left the file, so we had change history, but removed all of the settings since we didn't need any of them, as far as I know.

**Review by gsmet — APPROVED** [review]
